flogger/flogger-testing/src/main/java/com/google/common/flogger/testing/FakeLoggerBackend.java

```diff
@@ -64,6 +66,28 @@ public LogData getLogged(int n) {
     return logged.get(n);
   }
 
+  /**
+   * Return all captured {@link LogData}s.
+   */
+  public List<LogData> getLogged() {
+    return unmodifiableLogged;
+  }
+
+  /**
+   * Returns the last {@link LogData} entry captured by this backend.
+   */
+  public LogData getLastLogged() {
```

Comment:
To me, both this method and `getFirstLogged` should return `@Nullable` values, as the collection of `logged` may be empty. Right now, we would have `ArrayIndexOutOfBounds...` exceptions.

Also, I would get rid of `get` prefixes in Java code.

Reply:
I've made them throw an exception with the message instead of bare `ArrayIndexOutOfBounds`.

If targeting Kotlin in the first place (`FakeBackend` will also be migrated some day), `@Nullable` annotation pollute calls in Kotlin code with `!!` assertions. Lack of `get...()` prefixes eliminates possibility to call them as properties.